src/App/Fossa/DebugDir.hs

```diff
@@ -0,0 +1,23 @@
+module App.Fossa.DebugDir (
+  DebugDirRef,
+  newDebugDirRef,
+  readDebugDir,
+  writeDebugDir,
+) where
+
+import Data.IORef (IORef, newIORef, readIORef, writeIORef)
+
+-- | Type alias for the debug directory reference
+type DebugDirRef = IORef (Maybe FilePath)
```

Comment:
I'm not sure I understand how/why the `IORef` is needed. It seems like you pass the `IORef` around wherever it's needed.

Reply:
It's because the telemetry sink needs to know the directory name so that it can write telemetry.json to it, but the telemetry sink is instantiated before runSubCommand creates the directory. I tried a few ways to make it happen without this, but nothing felt great.

I'm definitely open to a different solution, though